dev/NavigationView/NavigationView.cpp

```diff
@@ -4774,7 +4774,7 @@ template<typename T> T NavigationView::GetContainerForData(const winrt::IInspect
     auto itemIndex = GetIndexFromItem(mainRepeater, data);
     if (itemIndex >= 0)
     {
-        if (auto container = mainRepeater.TryGetElement(itemIndex))
+        if (auto container = mainRepeater.GetOrCreateElement(itemIndex))
```

Comment:
>GetOrCreateElement [](start = 42, length = 18)

force creating these elements might not be the correct solution. This will end up being the anchor and layout will start to revolve around this element now. Could you instead wait until the elements are loaded and then set selection ?

Reply:
I realized that after I submitted the PR, see my comment above. I'll rework the solution to check if I should set selection in OnElementPrepared.

Reply:
Updated to do this. Note that the selectionmodel wasn't updated properly either because it also relied on elements being realized so I updated that as well.